### DISCUSSION TITLE
ES module project usage： "type": "module" which declares all .js files in that package scope as ES modules

### DISCUSSION THREAD

**Discussion by yuzheng14** [Zhengqbbb/cz-git#79] [Q&A] [answered]
完整报错为

```
cz-cli@4.2.5, cz-git@1.3.12

require() of ES Module /home/yuzheng14/code/vue/test/.commitlintrc.js from /home/yuzheng14/code/vue/test/node_modules/cz-git/lib/index.js not supported.
.commitlintrc.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename .commitlintrc.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /home/yuzheng14/code/vue/test/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).
```

看了下源码似乎是使用的 require() 进行导入，但是文档里模板给的是 .commitlintrc.js，这样不是压根没法用吗？

**Comment by Zhengqbbb** [comment, marked as answer]
> 文档里模板给的是 .commitlintrc.js

文档里面还有 `.commitlintrc.cjs` 和 `commitlint.config.cjs` 可以试试

if u project is esm project type: module. u can use .cjs configure like commitlint.config.cjs| .commitlintrc.cjs | cz.config.cjs

**Reply by yuzheng14** [reply]
原来如此，我在 `commitlint` 官方文档里没有搜到 `.commitlintrc.cjs` 这种格式还以为不支持

**Comment by Zhengqbbb** [comment]
Hi there, I plan to add support for loading ESM configuration files in v1.11.0 next weekend.
I have released a next pre-release version 🌟(v1.11.0-beta.1)🌟 for testing.
Welcome to try it out! 👀 Let me know if there are still any issues. 🎉

Docs preview link: https://deploy-preview-197--cz-git.netlify.app/config/
Example: https://github.com/Zhengqbbb/czgit-playground/tree/commitizen-esm

**Comment by Zhengqbbb** [comment]
Hi there 👋
The latest version 🌟(v1.11.0)🌟 released and supported ESM configure loading.
Docs Link: https://cz-git.qbb.sh/config/
💪 Please have a try. 👀 Let me know if there are still any issues. 🎉
